File: Python/gadget.py
# gadget.py
from electronic import Electronic

class Gadget(Electronic):
    def __init__(self, id, nama, harga, merk, stok, garansi, model, kategori, kapasitas):
        super().__init__(id, nama, harga, merk, stok, garansi)
        self.model = model
        self.kategori = kategori
        self.kapasitas = kapasitas

    def to_row(self):
        return [self.id, self.nama, str(self.harga), self.merk, str(self.stok), str(self.garansi),
                self.model, self.kategori, str(self.kapasitas)]

File: JAVA/Gadget.java
public class Gadget {
    private String id, nama, merk, model, kategori;
    private int harga, stok, garansi, kapasitas;

    public Gadget(String id, String nama, int harga, String merk, int stok,
                  int garansi, String model, String kategori, int kapasitas) {
        this.id = id;
        this.nama = nama;
        this.harga = harga;
        this.merk = merk;
        this.stok = stok;
        this.garansi = garansi;
        this.model = model;
        this.kategori = kategori;
        this.kapasitas = kapasitas;
    }

    // Getter dan Setter
    public String getId() { return id; }
    public int getStok() { return stok; }
    public void setStok(int stok) { this.stok = stok; }

    // Method untuk tabel
    public String[] toRowParts() {
        return new String[]{id, nama, String.valueOf(harga), merk,
                            String.valueOf(stok), String.valueOf(garansi),
                            model, kategori, String.valueOf(kapasitas)};
    }
}
